### ISSUE TITLE
`RandomGraph` Default Behavior

### ISSUE THREAD

**Issue by jeremy-wayland** [aidos-lab/rings#8]
## Bug / Design Issue: RandomGraph Example in Docs Fails Assertion

Our current documentation example for `RandomGraph` fails the expected assertion that the number of edges remains the same after transformation:

<details>
<summary>Example code from docs</summary>

```python
import torch
from torch_geometric.data import Data
from rings.perturbations import RandomGraph

# Create a simple graph
edge_index = torch.tensor([[0, 1, 1, 2], [1, 0, 2, 1]], dtype=torch.long)
data = Data(edge_index=edge_index, num_nodes=4)

# Example 1: Random graph with same number of edges
torch.manual_seed(42)  # For reproducibility
transform1 = RandomGraph()
t1_data = transform1(data.clone())
# Should have same number of edges but different structure
assert t1_data.edge_index.shape[1] == data.edge_index.shape[1]
assert not torch.equal(t1_data.edge_index, data.edge_index)
```

</details>

## Observed behavior

The first assertion fails because the transformed graph can have fewer edges. This is because our default Erdos-Rényi implementation drops self-loops, which reduces the number of edges.

Relevant code:
```python
# Remove self-loops
mask = row != col
row, col = row[mask], col[mask]
```

## Proposed solution
Refactor the self-loop removal logic, which is currently duplicated / ad hoc in places:
	•	Extract `_ensure_no_self_loops` into a standalone utility function.
	•	Reuse this utility consistently in perturbations.py, including `RandomGraph`, `ShuffleTransform`, and maybe future uses.
